packages/blockchain-wallet-v4-frontend/src/scenes/Nfts/AssetViewOnly/AssetViewOnly.tsx

```diff
@@ -0,0 +1,451 @@
+                          <BlockchainIcon
+                            cursor
+                            data-e2e='close'
+                            name='close'
+                            size='20px'
+                            color='grey600'
+                            role='button'
+                            onClick={() => goToView()}
+                          />
+                        </ItemDetails>
+                      )}
+                      <CollectionHeader>
+                        <div>
+                          <Text
+                            size='14px'
+                            weight={600}
+                            style={{
+                              alignItems: 'center',
+                              padding: '0em 0em 0.5em 0em'
+                            }}
+                          >
+                            Collection
+                          </Text>
+                          <div style={{ padding: '6px 0px' }}>
+                            <ShadowTag>
+                              <CollectionName>
+                                {asset?.image_url ? (
+                                  <NftCollectionImage
+                                    alt='Dapp Logo'
+                                    src={asset?.image_url || ''}
+                                    isVerified={false}
+                                    // isVerified={asset?.safelist_request_status === 'verified'}
+                                  />
+                                ) : (
+                                  <Flex alignItems='center' flexDirection='row'>
+                                    <Avatar
+                                      size={34}
+                                      name={asset?.image_url || ''}
+                                      variant='marble'
+                                      colors={AvatarGradientColors}
+                                    />
+                                  </Flex>
+                                )}
+                                <Text
+                                  size='16px'
+                                  weight={600}
+                                  style={{
+                                    maxWidth: '250px',
+                                    overflow: 'hidden',
+                                    paddingLeft: '8px',
+                                    textOverflow: 'ellipsis',
+                                    whiteSpace: 'nowrap'
+                                  }}
+                                >
+                                  {asset?.asset_contract.name}
+                                </Text>
+                              </CollectionName>
+                            </ShadowTag>
+                          </div>
+                        </div>
+                      </CollectionHeader>
+                      <AssetName>
+                        {asset?.token_id && asset?.token_id.length > 12
```

Comment:
you don't really need this length check here. You could just do `asset?.token_id.substring(0, 12)` and it should work like you want.